tests/nvmeof/test_ceph_nvmeof_data_integrity.py

```diff
@@ -0,0 +1,238 @@
+            gateway.delete_subsystem(subnqn=sub_cfg["nqn"])
+
+        if "gateway" in config["cleanup"]:
+            delete_gateway(gw_node)
+
+    if "pool" in config["cleanup"]:
+        rbd_obj.clean_up(pools=[config["rbd_pool"]])
+
+
+def run(ceph_cluster: Ceph, **kwargs) -> int:
+    """Return the status of the Ceph NVMEof test execution.
+
+    - Configure SPDK and install with control interface.
+    - Configures Initiators and Run FIO and test data integrity on NVMe targets.
+    - Cleanup the ceph-nvme gw entities.
+
+    Args:
+
+        ceph_cluster: Ceph cluster object
+        kwargs: Key/value pairs of configuration information to be used in the test.
+
+    Returns:
+        int - 0 when the execution is successful else 1 (for failure).
+
+    Example:
+
+
+    """
+    LOG.info("Starting Ceph Ceph NVMEoF deployment.")
+
+    config = kwargs["config"]
+    rbd_pool = config["rbd_pool"]
+    rbd_obj = initial_rbd_config(**kwargs)["rbd_reppool"]
+
+    if config.get("cleanup-only"):
+        teardown(ceph_cluster, rbd_obj, config)
+        return 0
+
+    try:
+        gw_node = get_node_by_id(ceph_cluster, config["gw_node"])
+
+        gateway = Gateway(gw_node)
+        if config.get("install"):
+            configure_spdk(gw_node, rbd_pool)
+
+        if config.get("subsystems"):
+            with parallel() as p:
+                for subsystem in config["subsystems"]:
+                    p.spawn(configure_subsystems, rbd_obj, rbd_pool, gateway, subsystem)
+
+        if config.get("initiator"):
+            targets = json.loads(initiators(ceph_cluster, gateway, config["initiator"]))
+            LOG.info(f"Targets discovered: {targets}")
+
+            # verifying data integrity on NVMe targets
+            for target in targets["Devices"]:
+                nvme = generate_unique_id(length=4)
+                mount_point = f"/tmp/{nvme}"  # Change this to the desired mount point
+                create_filesystem_and_mount(rbd_obj, target["DevicePath"], mount_point)
+                # FIO command to run on NVMe targets
+                fio_command = (
+                    f"fio --ioengine=sync --refill_buffers=1 "
+                    f"--log_avg_msec=1000 --size=1g --bs=128k "
+                    f"--filename={mount_point}/test.txt"
+                )
+
+                # Run FIO for Writing the files
+                LOG.info("Running FIO for write operations.")
+                cmd_write = f"{fio_command} --name=writeiops --rw=randwrite"
+                rbd_obj.exec_cmd(cmd=cmd_write)
+                # Calculate md5sum after FIO write
+                md5sum_write = rbd_obj.exec_cmd(cmd=f"md5sum {mount_point}/test.txt")
+                # Run FIO for reading the files
+                LOG.info("Running FIO for read operations.")
+                cmd_read = f"{fio_command} --name=readiops --rw=randread"
+                rbd_obj.exec_cmd(cmd=cmd_read)
+                # Calculate md5sum after FIO read
```

Comment:
Can we have some operations done before comparing the md5sum? 
Ex: Network restarts,service restarts,data unavilability (unmounts) scenarios.

Reply:
Yes added unmounts before comparing - Will take other scenarios as separate test case